tests/api2/test_530_ups.py

```diff
@@ -28,25 +36,92 @@
     'monpwd'
 ]
 
+default_dummy_data = {
+    'battery.charge': 100,
+    'driver.parameter.pollinterval': 2,
+    'input.frequency': 49.9,
+    'input.frequency.nominal': 50.0,
+    'input.voltage': 230,
+    'input.voltage.nominal': 240,
+    'ups.status': 'OL',
+    'ups.timer.shutdown': -1,
+    'ups.timer.start': -1,
+}
+
+
+def remove_file(filepath):
+    ssh(f'rm {filepath}', check=False)
+
+
+def did_shutdown():
+    return ssh(f'cat {SHUTDOWN_MARKER_FILE}', check=False) == "done\n"
+
+
+def write_fake_data(data={}):
+    all_data = default_dummy_data | data
+    with NamedTemporaryFile() as f:
+        for k, v in all_data.items():
+            f.write(f'{k}: {v}\n'.encode('utf-8'))
+        f.flush()
+        os.fchmod(f.fileno(), 0o644)
+        results = send_file(f.name, DUMMY_FAKEDATA_DEV, user, password, truenas_server.ip)
+        assert results['result'], str(results['output'])
+
+
+def wait_for_alert(klass, retries=10):
+    assert retries > 0
+    while retries:
+        alerts = call('alert.list')
+        for alert in alerts:
+            if alert['klass'] == klass:
+                return alert
+
+
+@pytest.fixture(scope='module')
+def ups_running():
+    with ensure_service_enabled('ups'):
+        with ensure_service_started('ups'):
+            yield
+
+
+@pytest.fixture(scope='module')
+def dummy_ups_driver_configured():
+    write_fake_data()
+    remove_file(SHUTDOWN_MARKER_FILE)
+    old_config = call('ups.config')
+    del old_config['complete_identifier']
+    del old_config['id']
+    payload = {
+        'mode': 'MASTER',
+        'driver': 'dummy-ups',
+        'port': DUMMY_FAKEDATA_DEV,
+        'description': 'dummy-ups in dummy-once mode',
+        'shutdowncmd': f'echo done > {SHUTDOWN_MARKER_FILE}'
+    }
+    call('ups.update', payload)
+    try:
+        yield
+    finally:
+        call('ups.update', old_config)
+        remove_file(SHUTDOWN_MARKER_FILE)
+        remove_file(DUMMY_FAKEDATA_DEV)
+
 
 def test_01_get_ups_service_id():
     global ups_id
-    results = GET('/service/?service=ups')
-    assert results.status_code == 200, results.text
-    assert results.json()[0]['state'] == 'STOPPED', results.text
-    assert results.json()[0]['enable'] is False, results.text
-    ups_id = results.json()[0]['id']
+    results = call('service.query', [['service', '=', 'ups']], {'get': True})
+    assert results['state'] == 'STOPPED', results
+    assert results['enable'] is False, results
+    ups_id = results['id']
 
 
 def test_02_Enabling_UPS_Service_at_boot():
```

Comment:
`test_02_` and `test_03` aren't valuable imo. We need to fix tests like this (there are a bunch in this file). Even more so because you've imported context managers `ensure_service_enabled` and `ensure_service_started`. These types of "tests" can be removed entirely because you've got other areas in the code that will fail if the service isn't enabled/started. If you disagree, then we should combine tests like this into 1 test function.
```
def test_enable_service():
    call('service.update', blah...)
    rv = get_service_state()
    assert rv['enable'], rv